### ISSUE TITLE
Support for setting `secrets` as a map

### ISSUE THREAD

**Issue by abstrctn** [oliver006/drone-cloud-run#16]
Google now [has preview support](https://cloud.google.com/run/docs/configuring/secrets) for setting environment variable values from secrets in Secret Manager. On the command line, it works very similarly to environment variables, with [the same suite of](https://cloud.google.com/sdk/gcloud/reference/run/deploy#--clear-secrets) `--clear-secrets` / `--set-secrets` / `--remove-secrets` and `--update-secrets`.

We've been using it so far in by passing `--set-secrets` as a flag in `addl_flags`, but it becomes challenging when setting multiple variables, since we wind up with a long comma-delimited string that's hard to parse by humans. Given that secrets are set as key-value pairs, it seems like it would be more natural to use a map like we currently do with the `environment` option.

There are several formats supported with secrets, letting you set either environment variables from secrets, or setting the contents of a filepath to a secret value. But both are still fundamentally a map that could be represented in yaml:

```yaml
secrets:
  ENV_1: secret_name:latest
  /mnt/path: secret_name:latest
```

I put together a PR to illustrate how we might go about this, but we can continue the discussion here. One potential conflict is that there already seems to be a top-level `secrets` array in the README example for Drone v0.8. I'm not familiar with the older syntax, so not sure whether that's a reserved key, or would cause problems in the newer version as well...

(There are [also config maps](https://cloud.google.com/sdk/gcloud/reference/run/deploy#--clear-config-maps) that might be better represented this way as well, but we haven't used them so far.)

**Comment by oliver006** [comment]
Thanks for raising this issue (and opening the PR)!
I'll have a look at this more in-depth tomorrow but I think the overall idea makes sense. And I'm fine with making backwards incompatible changes with regards to Drone 0.8 as long as we tag the plugin release with a new major version number.
